File: src/main/java/superstitio/delayHpLose/PreventHpLimit_Turns.java
package superstitio.delayHpLose;

import com.megacrit.cardcrawl.core.AbstractCreature;
import superstitio.DataManager;
import superstitio.powers.AbstractSuperstitioPower;

import static superstitio.delayHpLose.DelayHpLosePower_ApplyOnAttacked.IPreventHpLimit;

public class PreventHpLimit_Turns extends AbstractSuperstitioPower implements IPreventHpLimit {
    public static final String POWER_ID = DataManager.MakeTextID(PreventHpLimit_Turns.class);

    public PreventHpLimit_Turns(final AbstractCreature owner, int amount) {
        super(POWER_ID, owner, amount);
    }

    @Override
    public void updateDescriptionArgs() {
        this.setDescriptionArgs(amount);
    }

    @Override
    public void atEndOfRound() {
        addToBot_AutoRemoveOne(this);
    }

    @Override
    public void onPreventHpLimit() {
    }
}

File: superstitioMain/java/superstitio/delayHpLose/PreventHpLimit_Turns.kt
package superstitio.delayHpLose

import com.megacrit.cardcrawl.core.AbstractCreature
import superstitio.DataManager
import superstitio.delayHpLose.DelayHpLosePower_ApplyOnAttacked.IPreventHpLimit
import superstitio.powers.AbstractSuperstitioPower
import superstitioapi.utils.setDescriptionArgs

class PreventHpLimit_Turns(owner: AbstractCreature, amount: Int) : AbstractSuperstitioPower(POWER_ID, owner, amount),
    IPreventHpLimit
{
    override fun updateDescriptionArgs()
    {
        this.setDescriptionArgs(amount)
    }

    override fun atEndOfRound()
    {
        addToBot_AutoRemoveOne(this)
    }

    override fun onPreventHpLimit()
    {
    }

    companion object
    {
        val POWER_ID: String = DataManager.MakeTextID(PreventHpLimit_Turns::class.java)
    }
}
